test/integration/file_artifact_test.js

```diff
@@ -228,4 +231,67 @@ suite('artifact extration tests', () => {
     // Missing artifact should have an error...
     assert.equal(result.artifacts['public/my-missing.txt'].storageType, 'error');
   });
+
+  test('upload retry', async () => {
+    let worker = new TestWorker(DockerWorker);
+    await worker.launch();
+
+    worker.on('task run', function() {
+      iptables.reject({
+        chain: 'OUTPUT',
+        protocol: 'tcp',
+        dport: 443,
+        sudo: true
+      });
+    });
+
+    let retry = false;
+
+    worker.on('retrying artifact upload', function() {
```

Comment:
Hrm, will this alter the iptables for the host that it's running on? I wonder if we could do this just for the worker container that starts up.